pkg/master/reconcilers/lease.go

```diff
@@ -96,6 +98,11 @@ func (s *storageLeases) UpdateLease(ip string) error {
 	})
 }
 
+// RemoveLease removes the lease on a master IP in storage
+func (s *storageLeases) RemoveLease(ip string) error {
```

Comment:
can you split the introduction of the pre-stop-hook from the lease logic into two commits? They are not related.

Reply:
Sure